### PR TITLE
Propagate build info from BAR to VMR outputs

### PR DESCRIPTION
**Pull request by dkurepa**
https://github.com/dotnet/arcade-services/issues/4166
https://github.com/dotnet/arcade-services/issues/4165

This PR makes it so the VMR flows BAR builds. Flowing commits from local repositories won't work anymore.
This is done primarily by changing the `VmrDependencyUpdate` to get most of its information from the build.
The PR also removes package versions, PreReleaseVersionLabel, and `IsStable` properties from the git-info

### CODE REVIEW

File: src/Microsoft.DotNet.Darc/Darc/Operations/VirtualMonoRepo/InitializeOperation.cs [hunk 1]
```diff
@@ -34,8 +40,8 @@ protected override async Task ExecuteInternalAsync(
         CancellationToken cancellationToken)
         => await _vmrInitializer.InitializeRepository(
             repoName,
-            targetRevision,
-            null,
+            (await _barClient.GetBuildsAsync(repoName, targetRevision)).SingleOrDefault()
+                ?? throw new Exception($"Failed to, or found multiple builds for {repoName}/{targetRevision}"),
```
Comment (premun) [inline src/Microsoft.DotNet.Darc/Darc/Operations/VirtualMonoRepo/InitializeOperation.cs:44]: The message seems a bit weird to be honest. I would rather rephrase it to something like "Expected 1 build to exist for commit X of repo Y" or similar.

File: test/Microsoft.DotNet.Darc.VirtualMonoRepo.E2E.Tests/VmrSyncRepoChangesTest.cs [hunk 2]
```diff
@@ -57,41 +57,6 @@ public async Task FileChangesAreSyncedTest()
         await GitOperations.CheckAllIsCommitted(VmrPath);
     }
 
-    [Test]
-    public async Task PackageVersionIsUpdatedOnlyTest()
```
Comment (premun) [inline test/Microsoft.DotNet.Darc.VirtualMonoRepo.E2E.Tests/VmrSyncRepoChangesTest.cs:61]: We had this functionality so that it's possible to flow a new build of the same SHA. This was a requirement of the SourceBuild team. In those cases, we would only update the version files and be done.
I think this should still be supported.

**Review by premun — CHANGES_REQUESTED** [review]
I don't think we can change the API so that you cannot call `darc vmr update` for an arbitrary commit that does not have a build associated. This was not the plan.

We can forbid this functionality for the forward/backward commands but not for `darc vmr update` and `darc vmr initialize`. These commands are used by devs to manually flow code to repos right now and they are used by many scripts, devcontainers, pipelines and more.

I thought that we would basically only query for the information instead of computing it.
What's preventing that?

**Comment by dkurepa** [comment]
> I don't think we can change the API so that you cannot call `darc vmr update` for an arbitrary commit that does not have a build associated. This was not the plan.
> 
> We can forbid this functionality for the forward/backward commands but not for `darc vmr update` and `darc vmr initialize`. These commands are used by devs to manually flow code to repos right now and they are used by many scripts, devcontainers, pipelines and more.
> 
> I thought that we would basically only query for the information instead of computing it. What's preventing that?

Nothing is changing, this is still a supported scenario. However the only change here is that this will only work for builds that have been published to bar. So if you do something like `darc vmr update ...` if there is a build for that repo/commit, things will still work the same. The only thing that won't work is flowing local code flow. I think this was the agreement?

**Comment by premun** [comment]
> > I don't think we can change the API so that you cannot call `darc vmr update` for an arbitrary commit that does not have a build associated. This was not the plan.

> Nothing is changing, this is still a supported scenario.

Pick one :)

> I think this was the agreement?

I think the ask was to flow the build ID into the version files, nothing more. I really don't think we should change the `VmrUpdater` this much. It's used in many flows which have nothing to do with BAR. As I said above - these commands are used by devs to manually flow code to repos/VMR right now and they are used by many scripts, devcontainers, pipelines and more.
These are random commits on dev branches. For instance Matt uses it to test changes from repos in the VMR before submitting a PR.

**Comment by dkurepa** [comment]
as discussed offline, closing this PR to start over with a reduced scope that keeps the same usage for VmrUpdate and VmrInitialize
